internal/cluster/validator.go

```diff
@@ -528,177 +379,127 @@ func checkCidrsOverlapping(cluster *common.Cluster) error {
-		return "A mismatch between the number of Cluster and Service networks"
-	default:
-		return fmt.Sprintf("Unexpected status %s", status)
-	}
-}
+func (v *clusterValidator) isNetworksSameAddressFamilies(c *clusterPreprocessContext) (ValidationStatus, string) {
+	var status ValidationStatus
 
-func (v *clusterValidator) isNetworksSameAddressFamilies(c *clusterPreprocessContext) ValidationStatus {
-	//If one of the required Cidr fields is empty return Pending status
-	if !validationStatusToBool(v.isClusterCidrDefined(c)) || !validationStatusToBool(v.isServiceCidrDefined(c)) {
-		return ValidationPending
-	}
-	machineNetworkRequired := network.IsMachineNetworkRequired(c.cluster)
-	if machineNetworkRequired && !validationStatusToBool(v.isMachineCidrDefined(c)) {
-		return ValidationPending
-	}
-	serviceNetworkFamilies, err := network.CidrsToAddressFamilies(network.GetServiceNetworkCidrs(c.cluster))
-	if err != nil {
-		v.log.WithError(err).Errorf("Getting service address families for cluster %s", c.cluster.ID.String())
-		return ValidationError
-	}
-	clusterNetworkFamilies, err := network.CidrsToAddressFamilies(network.GetClusterNetworkCidrs(c.cluster))
-	if err != nil {
-		v.log.WithError(err).Errorf("Getting cluster address families for cluster %s", c.cluster.ID.String())
-		return ValidationError
-	}
-	clusterNetworkFamilies = network.CanonizeAddressFamilies(clusterNetworkFamilies)
-	if !reflect.DeepEqual(serviceNetworkFamilies, clusterNetworkFamilies) {
-		return ValidationFailure
-	}
-	if machineNetworkRequired {
-		machineNetworkFamilies, err := network.CidrsToAddressFamilies(network.GetMachineNetworkCidrs(c.cluster))
-		if err != nil {
-			v.log.WithError(err).Errorf("Getting machine address families for cluster %s", c.cluster.ID.String())
-			return ValidationError
-		}
-		machineNetworkFamilies = network.CanonizeAddressFamilies(machineNetworkFamilies)
-		if !reflect.DeepEqual(serviceNetworkFamilies, machineNetworkFamilies) {
-			return ValidationFailure
-		}
-	}
-	return ValidationSuccess
-}
+	var clusterCidrDefined ValidationStatus
+	clusterCidrDefined, _ = v.isClusterCidrDefined(c)
 
-func (v *clusterValidator) printIsNetworksSameAddressFamilies(context *clusterPreprocessContext, status ValidationStatus) string {
-	switch status {
-	case ValidationFailure:
-		var networks string
-		if network.IsMachineNetworkRequired(context.cluster) {
-			networks = "(MachineNetworks, ServiceNetworks, ClusterNetworks)"
+	var serviceCidrDefined ValidationStatus
+	serviceCidrDefined, _ = v.isServiceCidrDefined(c)
+
+	//If one of the required Cidr fields is empty return Pending status
+	if !validationStatusToBool(clusterCidrDefined) || !validationStatusToBool(serviceCidrDefined) {
+		status = ValidationPending
+	} else {
+		machineCidrDefined, _ := v.isMachineCidrDefined(c)
+		machineNetworkRequired := network.IsMachineNetworkRequired(c.cluster)
+		if machineNetworkRequired && !validationStatusToBool(machineCidrDefined) {
+			status = ValidationPending
```

Comment:
I think you could probably combine this with the message handling and return early, no?

`network.IsMachineNetworkRequired` is checking user managed networking so it'd imagine it's more accurate than the check below.